File: RecipeFinalAttempt/app/src/main/java/com/example/recipefinalattempt/RecipeActivity.java
package com.example.recipefinalattempt;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class RecipeActivity extends AppCompatActivity {
    private TextView mRecipeName;
    private TextView mRecipeIngredients;
    private TextView mRecipeMethodTitle;
    private TextView mRecipe;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipe);


        mRecipeName = findViewById(R.id.text_recipe);
        mRecipeIngredients = findViewById(R.id.ingredients);
        mRecipeMethodTitle = findViewById(R.id.method);
        mRecipe = findViewById(R.id.recipe);

        Intent intent = getIntent();
        String Title = intent.getExtras().getString("RecipeName");
        String Ingredients = intent.getExtras().getString("RecipeIngredients");
        String MethodTitle = intent.getExtras().getString("RecipeMethodTitle");
        String Recipe = intent.getExtras().getString("Recipe");

        if (Title != null) mRecipeName.setText(Title);
        if (Ingredients != null) mRecipeIngredients.setText(Ingredients);
        if (MethodTitle != null) mRecipeMethodTitle.setText(MethodTitle);
        if (Recipe != null) mRecipe.setText(Recipe);

    }
}

File: RecipeApp3/app/src/main/java/com/example/recipeapp/RecipeActivity.kt
package com.example.recipeapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class RecipeActivity : AppCompatActivity() {
    private lateinit var dishName: TextView
    private lateinit var category: TextView
    private lateinit var servings: TextView
    private lateinit var difficulty: TextView
    private lateinit var ingredients: TextView
    private lateinit var steps: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipe)
        dishName = findViewById(R.id.dishName)
        category = findViewById(R.id.category)
        servings = findViewById(R.id.servings)
        difficulty = findViewById(R.id.difficulty)
        ingredients = findViewById(R.id.ingredients)
        steps= findViewById(R.id.steps)

        val intent = intent
        dishName.text = intent.getStringExtra("dishName")
        category.text = intent.getStringExtra("category")
        servings.text = intent.getStringExtra("servings")
        difficulty.text = intent.getStringExtra("difficulty")
        ingredients.text = intent.getStringExtra("ingredients")
        steps.text = intent.getStringExtra("steps")

    }
}